nidaqmx_examples/playrec.py

```diff
@@ -0,0 +1,83 @@
+"""Simultaneous read and write with NI USB-4431 or similar device."""
+
+import numpy as np
+import nidaqmx as ni
+from nidaqmx.constants import WAIT_INFINITELY
+
+
+def query_devices():
+  local = ni.system.System.local()
+  driver = local.driver_version
+
+  print(f'DAQmx {0}.{1}.{2}'.format(
+    driver.major_version, driver.minor_version, driver.update_version))
+
+  for device in local.devices:
+    print('Device Name: {0}, Product Category: {1}, Product Type: {2}'.format(
+      device.name, device.product_category, device.product_type))
+
+
+def playrec(
+  data, samplerate=96000, input_mapping=['Dev2/ai0', 'Dev2/ai1', 'Dev2/ai2'],
+  output_mapping=['Dev2/ao0']
+):
+  """Simultaneous playback and recording though NI device.
+
+  Parameters:
+  -----------
+  data: array_like, shape (nsamples, len(output_mapping))
+    Data to be send to output channels.  
+  samplerate: int
+    Samplerate
+  input_mapping: list of str
+    Input device channels
+  output_mapping: list of str
+    Output device channels
+
+  Returns
+  -------
+  ndarray, shape (nsamples, len(input_mapping))
+    Recorded data
+
+  """
+  max_out_range = 3.5 # output range of USB-4431
+  max_in_range = 10   # input range of USB-4431
+  max_outdata = np.max(np.abs(outdata))
+  if max_outdata > max_out_range:
+    raise ValueError(
+      f"outdata amplitude ({max_outdata:.2f}) larger than allowed range"
+      f"(+-{max_out_range:.2f}).")
+
+  outdata = np.asarray(outdata).T
+  nsamples = outdata.shape[1]
+
+  with ni.Task() as read_task, ni.Task() as write_task:
+    for o in output_mapping:
+      aochan = write_task.ao_channels.add_ao_voltage_chan(o)
+      aochan.ao_max = max_out_range
+      aochan.ao_min = -max_out_range
```

Comment:
these are input params on `add_ao_voltage_chan`: `min_val`, `max_val`

Reply:
I was wondering why these are not automatically set when passing a device to `add_ai_voltage_chan`. Seems like they must be passed explicitly.